### PR TITLE
test(core): add missing tests for prompts/utils.ts

### LINKED ISSUE #19940: test(core): add missing tests for prompts/utils.ts
### Description

`packages/core/src/prompts/utils.ts` exports three utility functions 
used in prompt construction but has no corresponding test file:

- `resolvePathFromEnv()` — resolves environment variable values as 
  paths or boolean switches
- `applySubstitutions()` — applies template variable replacements 
  to prompt strings
- `isSectionEnabled()` — checks if a prompt section is enabled via 
  environment variables

Other files in the same directory (`promptProvider.ts`, 
`prompt-registry.ts`, `mcp-prompts.ts`) all have corresponding 
test files.

### Proposed fix

Create `packages/core/src/prompts/utils.test.ts` with comprehensive 
tests covering all three exported functions, including edge cases 
(empty input, whitespace, case sensitivity, home directory resolution 
failure).

I'd be happy to submit a PR for this.

### PR DESCRIPTION
**Pull request by krrishverma1805-web**
## Summary

Adds comprehensive test coverage for `packages/core/src/prompts/utils.ts`, 
which previously had no corresponding test file despite containing 
three exported utility functions used in prompt construction.

Fixes #19940

## Changes Made

- Created `packages/core/src/prompts/utils.test.ts` with 27 tests 
  covering all three exported functions:

### `resolvePathFromEnv()` (14 tests)
- Undefined, empty, and whitespace-only input
- Boolean switch values (`true`, `false`, `1`, `0`)
- Case-insensitive switch handling
- Whitespace trimming
- Absolute path resolution
- Tilde (`~`) expansion to home directory
- Bare tilde resolution
- Graceful handling of home directory resolution failure

### `isSectionEnabled()` (8 tests)
- Unset environment variable (defaults to enabled)
- Explicit enable values (`1`, `true`)
- Explicit disable values (`0`, `false`)
- Case-insensitive key conversion
- Whitespace handling around values
- Non-standard truthy values

### `applySubstitutions()` (8 tests)
- `${AgentSkills}` single and multiple replacements
- `${SubAgents}` with Gemini 3 vs legacy snippets
- `${AvailableTools}` with and without tools
- Tool-specific `${toolName_ToolName}` variables
- Prompts with no substitution placeholders

## Testing

- [x] All new tests pass (`npx vitest run packages/core/src/prompts/utils.test.ts`)
- [x] Full core test suite passes (`npm test -w @google/gemini-cli-core`)
- [x] Follows existing test patterns (Vitest, `vi.stubEnv()`, 
  `vi.mock()` for dependencies)
- [x] Includes Apache-2.0 license header with correct year (2026)

## Context

This fills a test coverage gap identified by comparing source files 
to test files in `packages/core/src/prompts/`. All other files in 
this directory (`promptProvider.ts`, `prompt-registry.ts`, 
`mcp-prompts.ts`) have corresponding test files.

### CODE REVIEW

**Review by jacob314 — APPROVED** [review]
lgtm

File: packages/core/src/prompts/utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+});
+
+describe('isSectionEnabled', () => {
+  afterEach(() => {
+    vi.unstubAllEnvs();
+  });
+
+  it('should return true when the env var is not set', () => {
+    expect(isSectionEnabled('SOME_KEY')).toBe(true);
+  });
```
Comment (mrpmohiburrahman) [inline packages/core/src/prompts/utils.test.ts:168]: Here, the test assumes `GEMINI_PROMPT_SOME_KEY` won't exist. However, if a
developer happens to have set this in their local system environment:

```bash
export GEMINI_PROMPT_SOME_KEY=0
```

This test will fail locally.

While you correctly included `vi.unstubAllEnvs()` in the `afterEach` block, this
Vitest function only restores the environment back to its initial state _before_
the test started—it doesn't aggressively clear pre-existing variables.

To make this test block 100% bulletproof and isolated from global state leakage,
we should explicitly clear the variable inside the test itself (similar to how
you use `vi.stubEnv` in the other tests):

```typescript
it('should return true when the env var is not set', () => {
  vi.stubEnv('GEMINI_PROMPT_SOME_KEY', '');
  expect(isSectionEnabled('SOME_KEY')).toBe(true);
});
```

Let me know what you think.

File: packages/core/src/prompts/utils.test.ts [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+      isSwitch: true,
+      value: 'true',
+      isDisabled: false,
+    });
+  });
+
+  it('should resolve a regular path', () => {
+    const result = resolvePathFromEnv('/some/absolute/path');
+    expect(result.isSwitch).toBe(false);
+    expect(result.value).toBe('/some/absolute/path');
```
Comment (mrpmohiburrahman) [inline packages/core/src/prompts/utils.test.ts:127]: In here, the expected strings hardcode POSIX (`/`) forward slashes:

```typescript
// Line 127 and Line 135
expect(result.value).toBe('/some/absolute/path');
```

When `resolvePathFromEnv` ( line 125 ) runs on a Windows computer, it uses Node's `path.resolve` and `path.join` to return a proper Windows-style path (like `"C:\some\absolute\path"`). However, the test is trying to match that result
against the hardcoded string `"/some/absolute/path"`. Because of this mismatch,
the test will always fail on Windows.

To guarantee cross-platform compatibility without changing your test logic, we
should wrap the expected strings in `path.resolve()` and `path.join()`. This
delegates the slash-formatting to Node.js natively:

```typescript
import path from 'node:path';

// ...

it('should resolve a regular path', () => {
  // Dynamically resolves to C:\... on Windows and /... on POSIX
  expect(result.value).toBe(path.resolve('/some/absolute/path'));
});
```

Let me know what you think?

File: packages/core/src/prompts/utils.test.ts [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+    expect(result.isDisabled).toBe(false);
+  });
+
+  it('should resolve a tilde path to the home directory', () => {
+    const result = resolvePathFromEnv('~/my/custom/path');
+    expect(result.isSwitch).toBe(false);
+    expect(result.value).toContain('/mock/home');
+    expect(result.value).toContain('my/custom/path');
```
Comment (mrpmohiburrahman) [inline packages/core/src/prompts/utils.test.ts:135]: Same thing as my previous comment on line 127

**Review by jacob314 — APPROVED** [review]
lgtm

**Review by jacob314 — APPROVED** [review]
